src/main/java/pl/psnc/dei/ui/components/SearchResultsComponent.java

```diff
@@ -394,4 +406,31 @@ public void clear() {
         searchResults.clear();
         updateComponent();
     }
+
+
+    public void selectAll() {
+        resultsList.removeAll();
+        searchResults.getResults().forEach(searchResult -> resultsList.add(createResultComponent(searchResult, true)));
+        currentUserRecordSelection.clearSelectedRecords();
+        searchResults.getResults().forEach(r -> currentUserRecordSelection.addSelectedRecordId(r.getId()));
+    }
+
+    public void inverseSelection() {
+        List<SearchResult> notSelected = searchResults.getResults().stream()
```

Comment:
Why do you create all those lists? Why not make one iteration and do whatever is needed for a search result (if it is on the user record selection list then it should be removed from it and added to result list with false otherwise it should be added to user record selection list and added to result list with true)?

Reply:
Good idea, changed